### PR TITLE
[Refactor] refactor image prune flag

### LINKED ISSUE #1922: Make `image prune` remove dangling images by default
Currently `nerdctl image prune` requires `--all` to be specified. Since dangling image is supported, we should remove this restriction and remove dangling images by default and remove all unused if `--all` is specified.

`nerdctl image prune`: https://github.com/containerd/nerdctl/blob/main/docs/command-reference.md#whale-nerdctl-image-prune

dangling support: https://github.com/containerd/nerdctl/pull/1919

`docker image prune`: https://docs.docker.com/engine/reference/commandline/image_prune/

### PR DESCRIPTION
**Pull request by suyanhanx**
part of #1680 

Signed-off-by: suyanhanx <suyanhanx@gmail.com>

### CODE REVIEW

File: pkg/cmd/image/prune.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+			filteredImages = append(filteredImages, image)
+		}
+	} else {
+		for _, image := range imageList {
+			_, tag := imgutil.ParseRepoTag(image.Name)
+
+			if tag == "" {
+				filteredImages = append(filteredImages, image)
+			}
+		}
```
Comment (djdongjin) [inline pkg/cmd/image/prune.go:73]: I think you can reuse this (`filteredImages = filterDangling(imageList, true)`:

https://github.com/containerd/nerdctl/blob/67c2e11a12262d796da8336fb300f31a9adb6b9a/pkg/cmd/image/list.go#L126-L139

**Comment by djdongjin** [comment]
I'd suggest to split the PR to a refactor PR and a PR fixing #1922, since we should have extra test to validate that `image prune` only removes dangling images not unused images.

File: pkg/cmd/image/prune.go (hunk 1 above)
Comment (suyanhanx) [reply pkg/cmd/image/prune.go:73]: I see it. My logic was copied from there.

Comment (suyanhanx) [reply pkg/cmd/image/prune.go]: May need to move to imgutil.

**Comment by suyanhanx** [comment]
> I'd suggest to split the PR to a refactor PR and a PR fixing #1922, since we should have extra test to validate that `image prune` only removes dangling images not unused images.

Done with new PR #1935

**Review by djdongjin — APPROVED** [review]
LGTM, thanks

**Comment by Zheaoli** [comment]
Maybe we should resolve the conflict before the PR has been merged

**Comment by suyanhanx** [comment]
Rebased.

**Review by Zheaoli — APPROVED** [review]
LGTM, thanks

**Pull request merged**